packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -364,6 +379,20 @@ export function useSelectableCollection(options: SelectableCollectionOptions): S
     }
   }, [isVirtualized, scrollRef, manager.focusedKey]);
 
+  // Update the focused key if the collection isn't focused and the selected keys changes to something that doesn't contain the focused key.
+  // Don't do this for virtual focus collections because the user is never focused within the collection and thus we don't
+  // want to mistakenly change their focus position.
+  let lastSelectedKeys = useRef(manager.selectedKeys);
+  useEffect(() => {
+    if (!equalSets(lastSelectedKeys.current, manager.selectedKeys)) {
+      if (!manager.isSelected(manager.focusedKey) && !manager.isFocused && !shouldUseVirtualFocus) {
+        manager.firstSelectedKey && manager.setFocusedKey(manager.firstSelectedKey);
+      }
+    }
+
+    lastSelectedKeys.current = manager.selectedKeys;
+  }, [manager.focusedKey, manager.isFocused, manager.firstSelectedKey, manager.selectedKeys, shouldUseVirtualFocus]);
```

Comment:
We should discuss this some more. I omitted a "isControlled" check because I figured that if the `selectedKey` changes and focus isn't within the collection then `selectedKey` MUST be controlled. If we still want that as a explicit check, I could add an `isControlled` getter to useMultipleSelectionState but then I'll need to expose that via the selection manager or pass it into useSelectableCollection...